### PR TITLE
Adding test case for aggregate exception

### PR DESCRIPTION
**Pull request by RyanDawkins**
This PR is a remake of #142 to pull to master instead of the PassExceptionsOnTaskFaults branch.

### CODE REVIEW

**Comment by cleemullins** [comment]
Looking now. Thanks!

**Comment by RyanDawkins** [comment]
After pulling in your changes and the changes to TestAdapter by @drub0y , my test required no changes. Let me know if you notice anything off!

File: tests/Microsoft.Bot.Builder.Tests/Adapter_TestAdapterTests.cs [hunk 1]
```diff
@@ -154,5 +156,26 @@ await adapter
                     .AssertReply("three")
                 .StartTest();
         }
+
+        [TestMethod]
```
Comment (cleemullins) [inline tests/Microsoft.Bot.Builder.Tests/Adapter_TestAdapterTests.cs]: Isn't this supposed to be DataTestMethod?

Comment (cleemullins) [reply tests/Microsoft.Bot.Builder.Tests/Adapter_TestAdapterTests.cs]: The docs are not much help, unfortunatly:
https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.testtools.unittesting.datatestmethodattribute?view=mstest-net-1.2.0

Comment (cleemullins) [reply tests/Microsoft.Bot.Builder.Tests/Adapter_TestAdapterTests.cs]: Never mind. I tested the code. It worked fine as-is.

**Comment by cleemullins** [comment]
Thanks Ryan! I think this is a great addition to the tests. 

Already tonight, with the Exception improvements to the Test Adapter, I've caught bugs that previously were slipping through. Having the exceptions flow through correctly has been a huge help!

**Comment by RyanDawkins** [comment]
@cleemullins  Sweet. I'll try and monitor the issues if you post them.

**Pull request merged**